spot-contracts/contracts/vaults/RolloverVault.sol

```diff
@@ -404,6 +404,58 @@ contract RolloverVault is
         return notes;
     }
 
+    /// @notice Deposits the underlying asset from {msg.sender} into the perp and vault systems,
+    ///         and mints both perp tokens and notes.
+    /// @param amount The amount tokens to be deposited into the vault.
+    /// @return The amount of perps and vault notes minted.
+    function deposit2(uint256 amount) external nonReentrant whenNotPaused returns (uint256, uint256) {
+        if (amount <= 0) {
+            revert UnacceptableDeposit();
+        }
+
+        uint256 totalSupply_ = totalSupply();
+        uint256 tvlBefore = getTVL();
+
+        // get deposit bond
+        IBondController depositBond = perp.getDepositBond();
+        BondTranches memory bt = depositBond.getTranches();
+
+        // NOTE: the vault may already have seniors which
+        //       we have to ignore for the sake of this operation.
+        uint256 seniorBalBefore = bt.tranches[0].balanceOf(address(this));
+
+        // transfer underlying in
+        underlying.safeTransferFrom(_msgSender(), address(this), amount);
+
+        // tranches underlying and updates vault assets with fresh tranches
+        _tranche(depositBond, bt, amount);
+
+        // mint perps using freshly tranched seniors
+        uint256 seniorsMinted = bt.tranches[0].balanceOf(address(this)) - seniorBalBefore;
+        _checkAndApproveMax(bt.tranches[0], address(perp), seniorsMinted);
+        uint256 perpsMinted = perp.deposit(bt.tranches[0], seniorsMinted);
+
+        // sync vault tranche balances
+        _syncAndRemoveDeployedAsset(bt.tranches[0]);
+        for (uint8 i = 1; i < bt.tranches.length; i++) {
+            _syncAndAddDeployedAsset(bt.tranches[i]);
+        }
+
+        // transfer perps back to user
+        IERC20Upgradeable(address(perp)).safeTransfer(_msgSender(), perpsMinted);
+        _syncAsset(perp);
+
+        // mint vault notes using remaning junior tranches
+        uint256 valueIn = getTVL() - tvlBefore;
+        uint256 notesMinted = (totalSupply_ > 0) ? totalSupply_.mulDiv(valueIn, tvlBefore) : (valueIn * INITIAL_RATE);
```

Comment:
should we move this to a helper function or pricing strategy, if we ever want to change how we price z's for minting?